### ISSUE TITLE
Error in docker image

### ISSUE THREAD

**Issue by ktkovachev** [sagan/EBWeb#6]
Hello, I have had trouble in trying to get the software to run on this machine; particularly, I can't get the docker image to work.
Whilst I have been able to get the program to run natively, albeit with `npm -i --force` (and it tells me there are many vulnerabilities?), both the pre-built and locally-built docker images I tried did not work, both having the same error:

```
0 verbose cli [ '/usr/local/bin/node', '/usr/local/bin/npm', 'start' ]
1 info using npm@8.3.1
2 info using node@v16.14.0
3 timing npm:load:whichnode Completed in 0ms
4 timing config:load:defaults Completed in 1ms
5 timing config:load:file:/usr/local/lib/node_modules/npm/npmrc Completed in 1ms
6 timing config:load:builtin Completed in 1ms
7 timing config:load:cli Completed in 1ms
8 timing config:load:env Completed in 0ms
9 timing config:load:file:/usr/src/app/.npmrc Completed in 1ms
10 timing config:load:project Completed in 1ms
11 timing config:load:file:/root/.npmrc Completed in 0ms
12 timing config:load:user Completed in 0ms
13 timing config:load:file:/usr/local/etc/npmrc Completed in 0ms
14 timing config:load:global Completed in 0ms
15 timing config:load:validate Completed in 1ms
16 timing config:load:credentials Completed in 1ms
17 timing config:load:setEnvs Completed in 0ms
18 timing config:load Completed in 6ms
19 timing npm:load:configload Completed in 6ms
20 timing npm:load:setTitle Completed in 1ms
21 timing config:load:flatten Completed in 2ms
22 timing npm:load:display Completed in 2ms
23 verbose logfile /root/.npm/_logs/2024-11-27T18_28_47_570Z-debug-0.log
24 timing npm:load:logFile Completed in 4ms
25 timing npm:load:timers Completed in 0ms
26 timing npm:load:configScope Completed in 0ms
27 timing npm:load Completed in 13ms
28 timing command:run-script Completed in 76820ms
29 timing command:start Completed in 76823ms
30 verbose stack Error: command failed
30 verbose stack     at ChildProcess.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/@npmcli/promise-spawn/index.js:64:27)
30 verbose stack     at ChildProcess.emit (node:events:520:28)
30 verbose stack     at maybeClose (node:internal/child_process:1092:16)
30 verbose stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:302:5)
31 verbose pkgid ebweb@1.2.18
32 verbose cwd /usr/src/app
33 verbose Linux 6.8.0-49-generic
34 verbose argv "/usr/local/bin/node" "/usr/local/bin/npm" "start"
35 verbose node v16.14.0
36 verbose npm  v8.3.1
37 error path /usr/src/app
38 error command failed
39 error signal SIGTERM
40 error command sh -c node index.js
41 verbose exit 1
42 timing npm Completed in 76972ms
43 verbose code 1
44 error A complete log of this run can be found in:
44 error     /root/.npm/_logs/2024-11-27T18_28_47_570Z-debug-0.log
```

Apparently there is some form of error in the NPM CLI module? I'm not particularly clear on it, though. The environment I run it under is as follows:

Docker version 26.1.3, build 26.1.3-0ubuntu1~24.04.1
`data-dicts` directory contains 広辞苑 (which runs fine natively) only
`config.js` is:
```
module.exports = {
  SITENAME: "EBWeb",
  CACHE_AGE: 86400,
  SHORT_CACHE_AGE: 120,
  MAX_DEFAULT: 100,
  MAX_CAP: 100,
  MAX_CAP2: 50,
  DEFAULTDICT: "", // The default dict. If not set, will use the first one in dicts list as default.
  JS_DOCUMENT_URL: "https://example.com/Custom%20JS%20guide",
  USERCONFIG_DOCUMENT_URL: "https://example.com/User%20Config%20guide",
  GAPI_CLIENTID: "", // "xxxxx-xxxxx.apps.googleusercontent.com"
  GAPI_KEY: "",
  GAPI_SECRET: "",
  GAPI_SCRIPT: "", // "https://script.google.com/macros/s/xxxxxxxxxx/exec"
  GOOGLE_SEARCH_API_KEY: "",
  GOOGLE_SEARCH_API_ENGINE: "",
  SITEID: "soradict", // Used as localStorage key. Do NOT change it in production site
  DATE: "2024-04-12",
  REDIRECT_NON_CANONICAL_URL: 0,
  PROXY: 0,
  OLD_STORAGE_KEY: "persist:state",
  KEYWORDS:
    "広辞苑,大辞林,日本国語大辞典,中日辞典,日中辞典,国語辞典,国語辞書,国語,辞書,辞典,意味,類語,例文,慣用句",
  DESCRIPTION: "当サイトでは、様々の大型辞典の内容を無料で閲覧できる",
  IBM_CLOUD_APIKEY: "", // for IBM translate API
  IBM_TRANSLATOR_API_URL: "", // "https://api.jp-tok.language-translator.watson.cloud.ibm.com/instances/fffff-fffff"
  ROOTPATH: "/", // The "pathname" part of (root) public URL of the site. Must re-rebuild Docker image if changed
  REALROOT: "/", // If program running behind a reverse proxy (eg. nginx), the root path that program sees
  PUBLIC_URL: "https://example.com", // Required. The "origin" part of public URL of the site
  IP: "0.0.0.0",
  PORT: 3000,
  HOME_HTML: `<div class="opt">
  <p>ようこそ。当サイトでは、様々の辞典の検索機能を無料に提供致します。</p>
</div>`,
  FOOTER_TEXT: `&copy;<a href="https://example.com/">example.com</a>`,
  HOME_FOOTER_TEXT: ``,
  EBCLIENT_BIN: "./binary/ebclient",
  EBCLIENT_DICT_PATH: "./data-dicts",
  //GA: "", // Set to enable google analytics. E.g.: "UA-1234567890-0"
  YAHOO_APPID: "", // yahoo.co.jp appid, for use Yahoo japanese morphological analyzer WebAPI
  DICTINFO: {
    // audioDict: "NHK日本語発音アクセント辞典",
    // audioDictEn: "研究社新英和中辞典",
    // kannjiDict: "学研漢和大字典",
    dicts: [
      {
        name: "広辞苑第六版",
        id: "広辞苑",
        alias: "",
        gaijiMap: "KOJIEN.json",
      },
      // {
      //   name: "ＮＨＫ　日本語発音アクセント辞典",
      //   id: "NHK日本語発音アクセント辞典",
      //   alias: "発音",
      //   gaijiMap: "",
      // },
    ], // dicts MUST be configured
    multiShortcuts: [
      // [
      //   "薦",
      //   "広辞苑_小学館中日・日中辞典_三省堂類語辞典",
      //   "管理人のお薦め辞典セットを使用する",
      // ],
    ], // 一括検索・ショートカット。each: [name, dicts, tip]
  },
};
```
...which again works fine natively.
The command I used to run the image was: `sudo docker run -d --name ebweb -v /root/ebweb/data-dicts:/usr/src/app/data-dicts -v /root/ebweb/config.js:/usr/src/app/config.js -p 127.0.0.1:3000:3000 ebweb`
(I also used the ebweb image from the container repository but that produced the same error.)
I do have the correct config.js and dictionary files installed in the mounted locations as well. Indeed, the log file says:
```> ebweb@1.2.18 start
> node index.js

dicts ["広辞苑第六版","付属資料"]
npm ERR! path /usr/src/app
npm ERR! command failed
npm ERR! signal SIGTERM
npm ERR! command sh -c node index.js

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2024-11-27T18_28_47_570Z-debug-0.log
```
which suggests that the dictionary was able to be loaded, but that another error was encountered later on.

Do you have any idea what is causing this error? Can it be easily fixed / is it a configuration error on my part? How can we address the many outdated packages that NPM flags up?

Thanks,
 - Kiril

**Comment by sagan** [comment]
Unfortunately I can't reproduce the same problem. Some advices:

* Check the `/root/.npm/_logs/*.log` file of the container, you may need to mount the `/root/.npm` volume.
* Check the log of docker engine.

I think the outdated dependencies do not post any serious security threat. I intend to continuously maintain this software, but I don't have much time right now. So please be patient.

**Comment by ktkovachev** [comment]
Hi, thanks so much for your time. I'm greatly sorry for wasting it, as it turns out, though. For one, for some reason, I also can't reproduce the bug from yesterday either. Also, I was having another issue connecting to the server (faced with repeated "connection reset" messages in the browser), which was fixed by restarting the host machine. I believe this had to do with my configuration of docker itself, since I added myself to the docker user group without fully restarting; which I am told might have been a source of the problem. At any rate, it works now. Thanks for your help!
